src/main/java/com/jwplayer/jwplatform/client/AdvertisingClient.java

```diff
@@ -0,0 +1,13 @@
+package com.jwplayer.jwplatform.client;
+
+public class AdvertisingClient{
+
+	private String path;
+	private String secret;
+	
+	public AdvertisingClient(String secret) {
+		this.secret = secret;
```

Comment:
Should be a precondition not null check on this also when moved to the static factory method.